proposals/11551-kubernetes-native-api/TestPlan.md

```diff
@@ -0,0 +1,87 @@
+| (K8s Mode) Create an experiment and a run                                                                                            |                                                                                                                                       |                                                                                                              |
+
+#### Without migration script
+Create few pipelines in DB mode and migrate it by recreating those via CRs
+
+| **Test Case**                                                                                                                                                                           | **Test Steps**                                          | **Expected Result**                                                                                                                                                 |
+|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| (KFP DB mode) Create pipeline and a pipeline version and re-create the pipeline & version using CRs in K8s mode                                                                         | Match the pipeline and pipeline version (DB vs K8s)     | Pipeline and pipeline version should match exactly except for ID?                                                                                                   |
+| (KFP DB mode) Create pipeline and 2 pipeline versions with same spec and and re-create the pipeline & version using CRs (in K8s mode)                                                   | Match the pipeline and pipeline versions (DB vs K8s)    | Pipeline and pipeline version should match exactly except for ID?                                                                                                   |
+| (KFP DB mode) Create pipeline and 2 pipeline versions with different specs and and re-create the pipeline & version using CRs (in K8s mode)                                             | Match the pipelines and pipeline versions (DB vs K8s)   | Pipeline and pipeline version should match exactly except for ID?                                                                                                   |
+| (KFP DB mode) Create pipeline and a pipeline version and and re-create the pipeline & version using CRs but with different **pipelineName** than the one in pipelineSpecs (in K8s mode) |                                                         | Should this be allowed? Will webhooks throw a validation error                                                                                                      |
```

Comment:
The pipeline name in the pipeline spec is used as a default value when uploading a pipeline but doesn't need to match the CR name. I think this is a cut candidate.